src/select/typeahead/TypeaheadSelect.tsx

```diff
@@ -0,0 +1,189 @@
+  shouldDisplaySelectedOptions = true,
+  shouldFilterOptionsByKeyword = true,
+  isDisabled,
+  shouldShowEmptyOptions = true,
+  canOpenDropdownMenu = true,
+  areOptionsFetching
+}: TypeaheadSelectProps) {
+  const [isMenuOpen, setMenuVisibility] = useState(false);
+  const [computedDropdownOptions, setComputedDropdownOptions] = useState(dropdownOptions);
+  const [shouldResetTypeaheadValue, setShouldResetTypeaheadValue] = useState(false);
+  const [shouldFocusOnInput, setShouldFocusOnInput] = useState(false);
+  const typeaheadSelectClassName = classNames(
+    "typeahead-select-dropdown",
+    customClassName
+  );
+  const tags = mapDropdownOptionsToTagShapes(selectedOptions);
+  const shouldDisplayOnlyTags = Boolean(
+    selectedOptionLimit && selectedOptions.length >= selectedOptionLimit
+  );
+  const canSelectMultiple = !selectedOptionLimit || selectedOptionLimit > 1;
+  const shouldCloseOnSelect =
+    !canSelectMultiple ||
+    Boolean(selectedOptionLimit && selectedOptions.length >= selectedOptionLimit - 1);
+  const typeaheadClassName = classNames("typeahead-select-header", {
+    "is-dropdown-menu-open": isMenuOpen,
+    "can-select-multiple": canSelectMultiple
+  });
+  const typeaheadInputRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    setComputedDropdownOptions(dropdownOptions);
+  }, [dropdownOptions]);
+
+  useEffect(() => {
+    let timeoutId: any;
+
+    if (shouldFocusOnInput) {
+      timeoutId = setTimeout(() => {
+        if (typeaheadInputRef.current) {
+          typeaheadInputRef.current.focus();
+          setShouldFocusOnInput(false);
+        }
+      });
+    }
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [shouldFocusOnInput]);
+
+  const dropdownHeader = (
+    <div className={"typeahead-select-header-container"}>
+      {shouldDisplaySelectedOptions && Boolean(tags.length) && (
+        <div className={"typeahead-select-header-tags-container"}>
+          {tags.map((tag, index) => (
+            <Tag
+              key={tag.id}
+              testid={`${testid}.tag-${index}`}
+              onRemove={handleRemove}
+              customClassName={"typeahead-select-tag"}
+              tag={tag}
+            />
+          ))}
+        </div>
+      )}
+
+      {!shouldDisplayOnlyTags && (
+        <TypeaheadInput
+          testid={`${testid}.search`}
+          customClassName={typeaheadClassName}
+          inputContainerRef={typeaheadInputRef}
+          id={typeaheadProps.id}
+          name={typeaheadProps.name}
+          placeholder={typeaheadProps.placeholder}
+          onQueryChange={handleKeywordChange}
+          shouldResetValue={shouldResetTypeaheadValue}
+          rightIcon={
+            areOptionsFetching ? (
+              <Spinner spinnerColor={"#EBEBEB"} backgroundColor={"white"} />
+            ) : (
+              <CaretDownIcon aria-hidden={true} />
+            )
+          }
+          onFocus={openDropdownMenu}
+        />
```

Comment:
```suggestion
           ...
           isDisabled={isDisabled}
           ...
        />
```

Although I am not sure, we should pass `isDisabled` as prop to `TypeaheadInput` to prevent dropdown menu to show up when it is disabled.